### PR TITLE
fix: cap bg opacity of hot lines for readability on dark themes

### LINKED ISSUE #56: Line heat color can make lines unreadable on dark themes
Hi! Great tools, thanks. I seem to have stumbled upon a minor problem with those line heat color decorations:

https://github.com/P403n1x87/austin-vscode/blob/d48825e62cb899d71846ae61d72052b9352187b3/src/view.ts#L46-L62

I'm using a dark theme and if there's a single hottest line, then it becomes kind of unreadable:

image

You can reproduce it by running the "Profile with Austin" action on this snippet:

```python
import time

time.sleep(5)
```

I tried other dark themes, including the default ones, and the line is more or less hard to read on most of them. While this is somewhat of a corner case, it would probably be nice to fix it regardless. What do you think?

The easiest solution I can think of is just `Math.min(opacity, 0.6)` it. The 60% opacity cap makes it at least (arguably) distinguishable enough:

image

60% is a big share of the time, so, if I understand it correctly, there can't even be more than one such line and it's safe to cap its color. In other words, we're not really missing anything by losing the ability to visually distinguish between lines in 60%-100% own% range :)

What's your thoughts on that?

### PR DESCRIPTION
**Pull request by iburakov**
A suggested way to fix #56

### CODE REVIEW

File: src/view.ts [hunk 1]
```diff
@@ -46,7 +46,7 @@ function formatMemory(bytes: number) {
 function setLineHeat(frame: FrameObject, own: number, total: number, overallTotal: number, localTotal: number, mode: string) {
     const editor = vscode.window.activeTextEditor;
     if (editor !== undefined) {
-        const opacity = own / localTotal;
+        const opacity = Math.min(own / localTotal, 0.6);
```
Comment (P403n1x87) [inline src/view.ts:49]: My worry here is that everything above 60% will have the same "heat". Perhaps we can use 0.6 as a factor, i.e.
```suggestion
        const opacity = 0.6 * own / localTotal;
```
This would probably make the smaller percentages barely visible. So perhaps the other adjustment could be a non-linear strictly monotonic mapping, e.g.
```suggestion
        const opacity = 0.6 * Math.sqrt(own / localTotal);
```

**Review by P403n1x87 — COMMENTED** [review]
Many thanks for your contribution! I have left some comments on the approach taken, and some potential alternatives. I'm not an expert in terms of colours, so mine are just some random ideas 🙂 .

File: src/view.ts (hunk 1 above)
Comment (iburakov) [reply src/view.ts:49]: Thanks for the prompt response! Yes, I share your concerns and was considering factor/non-linear approaches too. The sqrt is great for that use case, indeed.

It's just that then I noticed there can't be more than one line with `own / localTotal` > 0.6. I mean, the `localTotal` is just a sum of all local own times, so sum of all opacities must be 1.  If there's at most one line with "the same heat", then it's not really a problem :)

Is this right? Please correct me if I misunderstood something.

I may have failed to communicate that observation clear enough in the issue, sorry. Just making sure you got that.

Comment (iburakov) [reply src/view.ts:49]: Actually, the `sqrt` may be a just a better-suited mapping anyway, since making small %s more distinguishable seems to be important here in any case. It has a price though: hotter lines (x=0.5+) are more similar now:

image

What do you think in light of that? Going with sqrt?

Comment (P403n1x87) [reply src/view.ts:49]: > It's just that then I noticed there can't be more than one line with `own / localTotal` > 0.6

I'm not sure I follow. For example

```python
from time import sleep

sleep(1)
```

should have about 100% wall time on line 3. So there could be values that go above 60%, unless I misunderstood your example?

Comment (iburakov) [reply src/view.ts:49]: Right, in your example there's **one** such line that has ~100%. But there can't be **more than one such line** in a file.

This is because of how opacity is currently calculated. Here's a more visual example:
image
Own times here: `owns = [1, 3, 6, 0]`
`localTotal` will be [just their sum](https://github.com/P403n1x87/austin-vscode/blob/main/src/view.ts#L160): `localTotal = sum(owns)`
Opacities will be `owns[i] / localTotal` then: `[0.1, 0.3, 0.6, 0.0]`.

Sum of opacities must be 1, as you see, because they're calculated as `owns[i] / sum(owns)`.

So if there's one opacity that's `0.6`, all the rest **must** add up to `0.4` :) If there's an opacity of `0.8`, the rest must add up to `0.2`, etc.

In other words, there's **at most one** such hot line in a file, which must already have the brightest color in the file anyway, so comparing it to other lines by color isn't relevant, I assumed.

Comment (iburakov) [reply src/view.ts:49]: Actually, the more I think about it, the more I find doing the `sqrt` reasonable [due to this](https://github.com/P403n1x87/austin-vscode/pull/57#discussion_r1427764376), so let's just go with it.

**Comment by P403n1x87** [comment]
Thanks for your contribution! 🎉
